src/generator/type_inference_visitor.ts

```diff
@@ -1,32 +1,38 @@
 import {
     AssignmentExpression,
-    BinaryExpression, BooleanLiteral, CallExpression,
+    BinaryExpression,
+    BooleanLiteral,
+    CallExpression,
     Expression,
     FunctionDeclaration,
     Identifier,
-    isIdentifier,
-    LogicalExpression, MemberExpression,
-    NumericLiteral, UnaryExpression, VariableDeclarator,
+    isIdentifier, isMemberExpression,
+    LogicalExpression,
+    MemberExpression,
+    NumericLiteral,
+    UnaryExpression,
+    VariableDeclarator,
 } from '@babel/types';
 import Visitor from '../visitor';
 import {FunctionSignature, FunctionSignatures} from './generator';
-import {WebAssemblyType} from './wasm_type';
+import {getWebAssemblyType, WebAssemblyType} from './wasm_type';
 
 class TypeInferenceVisitor extends Visitor {
 
     private signatures: FunctionSignatures;
     private expressionTypes = new Map<Expression, WebAssemblyType>();
+    private variableTypes: WebAssemblyType[] = [];
```

Comment:
Maybe use a type declaration for this array type.

Reply:
The order in which the variable types are pushed to this array is important. Are you sure that it's always correct?